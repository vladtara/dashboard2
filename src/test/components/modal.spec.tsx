import { fireEvent, render } from "@testing-library/react";
import Modal, { IModalProps } from "../../components/modal";

const iconProps: IModalProps = {
  element: "icon",
  icon: "bug_report",
  title: "Test Modal",
  onClose: jest.fn(),
  children: <p>Hello</p>,
};

const invalidIconProps: IModalProps = {
  element: "icon",
  title: "Test Modal",
  onClose: jest.fn(),
  children: <p>Hello</p>,
};

const textProps: IModalProps = {
  element: "text",
  text: "Test",
  title: "Test Modal",
  onClose: jest.fn(),
  children: <p>Hello</p>,
};

const noneProps: IModalProps = {
  element: "none",
  title: "Test Modal",
  children: <p>Hello</p>,
};

const setup = (props: IModalProps) => {
  const modal = render(
    <Modal
      element={props.element}
      icon={props.icon}
      text={props.text}
      title={props.title}
      onClose={props.onClose}
    >
      {props.children}
    </Modal>,
  );

  return modal;
};

describe("modal.tsx", () => {
  it("Tests modal with icon button", () => {
    const modal = setup(iconProps);
    expect(modal.baseElement).toMatchSnapshot();
    fireEvent.click(modal.getByTestId("toggle-button"));
    fireEvent.click(modal.getByTestId("close-button"));
    expect(iconProps.onClose).toHaveBeenCalledTimes(2);
  });

  it("Tests modal with text button", () => {
    const modal = setup(textProps);
    expect(modal.baseElement).toMatchSnapshot();
    fireEvent.click(modal.getByTestId("toggle-button"));
    fireEvent.click(modal.getByTestId("close-button"));
    expect(textProps.onClose).toHaveBeenCalledTimes(2);
  });

  it("Tests modal with neither", () => {
    const modal = setup(noneProps);
    expect(modal.baseElement).toMatchSnapshot();
  });

  it("Tests modal with icon", () => {
    const modal = setup(invalidIconProps);
    expect(modal.baseElement).toMatchSnapshot();
    fireEvent.click(modal.getByTestId("toggle-button"));
    fireEvent.click(modal.getByTestId("close-button"));
    expect(invalidIconProps.onClose).toHaveBeenCalledTimes(2);
  });

  it("Tests modal without onClose behaviour", () => {
    const modal = setup(noneProps);
    expect(modal.baseElement).toMatchSnapshot();
    fireEvent.click(modal.getByTestId("close-button"));
    expect(iconProps.onClose).toHaveBeenCalledTimes(0);
  });
});
