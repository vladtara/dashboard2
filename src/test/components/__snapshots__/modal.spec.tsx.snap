// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal.tsx Tests modal with icon 1`] = `
<body>
  <div>
    <div
      class="sc-ftvSup cpmMIl"
      hidden=""
    >
      <div
        class="sc-jqUVSM izDwOx"
      >
        <h2
          class="sc-gsnTZi jiVNCT"
        >
          Test Modal
        </h2>
        <button
          class="sc-gKXOVf inaviU"
          data-testid="close-button"
        >
          <i
            class="sc-iBkjds fflOgg"
          >
            close
          </i>
        </button>
      </div>
      <p>
        Hello
      </p>
    </div>
    <button
      class="sc-gKXOVf inaviU"
      data-testid="toggle-button"
    >
      <i
        class="sc-iBkjds fflOgg"
      />
    </button>
  </div>
</body>
`;

exports[`modal.tsx Tests modal with icon button 1`] = `
<body>
  <div>
    <div
      class="sc-ftvSup cpmMIl"
      hidden=""
    >
      <div
        class="sc-jqUVSM izDwOx"
      >
        <h2
          class="sc-gsnTZi jiVNCT"
        >
          Test Modal
        </h2>
        <button
          class="sc-gKXOVf inaviU"
          data-testid="close-button"
        >
          <i
            class="sc-iBkjds fflOgg"
          >
            close
          </i>
        </button>
      </div>
      <p>
        Hello
      </p>
    </div>
    <button
      class="sc-gKXOVf inaviU"
      data-testid="toggle-button"
    >
      <i
        class="sc-iBkjds fflOgg"
      >
        bug_report
      </i>
    </button>
  </div>
</body>
`;

exports[`modal.tsx Tests modal with neither 1`] = `
<body>
  <div>
    <div
      class="sc-ftvSup cpmMIl"
      hidden=""
    >
      <div
        class="sc-jqUVSM izDwOx"
      >
        <h2
          class="sc-gsnTZi jiVNCT"
        >
          Test Modal
        </h2>
        <button
          class="sc-gKXOVf inaviU"
          data-testid="close-button"
        >
          <i
            class="sc-iBkjds fflOgg"
          >
            close
          </i>
        </button>
      </div>
      <p>
        Hello
      </p>
    </div>
  </div>
</body>
`;

exports[`modal.tsx Tests modal with text button 1`] = `
<body>
  <div>
    <div
      class="sc-ftvSup cpmMIl"
      hidden=""
    >
      <div
        class="sc-jqUVSM izDwOx"
      >
        <h2
          class="sc-gsnTZi jiVNCT"
        >
          Test Modal
        </h2>
        <button
          class="sc-gKXOVf inaviU"
          data-testid="close-button"
        >
          <i
            class="sc-iBkjds fflOgg"
          >
            close
          </i>
        </button>
      </div>
      <p>
        Hello
      </p>
    </div>
    <p
      class="sc-papXJ gkwCsw"
      data-testid="toggle-button"
    >
      Test
    </p>
  </div>
</body>
`;

exports[`modal.tsx Tests modal without onClose behaviour 1`] = `
<body>
  <div>
    <div
      class="sc-ftvSup cpmMIl"
      hidden=""
    >
      <div
        class="sc-jqUVSM izDwOx"
      >
        <h2
          class="sc-gsnTZi jiVNCT"
        >
          Test Modal
        </h2>
        <button
          class="sc-gKXOVf inaviU"
          data-testid="close-button"
        >
          <i
            class="sc-iBkjds fflOgg"
          >
            close
          </i>
        </button>
      </div>
      <p>
        Hello
      </p>
    </div>
  </div>
</body>
`;
